universal-application-tool-0.0.1/app/views/questiontypes/EnumeratorQuestionRenderer.java

```diff
@@ -118,10 +118,18 @@ private static Tag enumeratorField(
                     localizedEntityType))
             .addReferenceClass(ReferenceClasses.ENTITY_NAME_INPUT)
             .getContainer();
+    String confirmationMessage =
+        messages.at(MessageKey.ENUMERATOR_DIALOG_CONFIRM_DELETE.getKeyName(), localizedEntityType);
     Tag removeEntityButton =
         TagCreator.button()
             .withType("button")
             .withCondId(existingIndex.isPresent(), existingIndex.map(String::valueOf).orElse(""))
+            .attr(
+                "onclick",
+                String.format(
+                    "if(confirm('%s')){ return true; } else { var e = arguments[0] ||"
+                        + " window.event; e.stopImmediatePropagation(); return false; }",
+                    confirmationMessage))
```

Comment:
Nit. I've been trying to avoid writing JS within the Java layer, but the only other way I can think of doing it would be to add a hidden div just to ship the translation down to the client...

Reply:
This is not the most helpful second opinion (re your top-level comment), but I don't feel strongly one way or the other. I would say our JS in general scares me a bit, but having this one line of JS here doesn't bother me any extra.